.github/workflows/markdown-checks.yaml

```
@@ -0,0 +1,48 @@
+name: Markdown linter
+
+# What the execution context is:
+# The creation of a pull request. Can also be triggered manually.
+
+# What it does:
+# Performs Markdown linting on the content of the pull request.
+
+# Why we need it:
+# To maintain a level of consistency in the markup content.
+
+# What's important to know:
+# As well as the rules defined in the configuration, a custom
+# rule is also employed to check title case.
+
+on:
+  pull_request:
+    branches: [main]
```

You could also add an additional [`paths`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths) filter here which will prevent the workflow from triggering unless at least one file in the PR diff has a path = `docs/*`. This can help avoid clutter in your Actions log since changes to e.g. `README.md` will not trigger a workflow run.
```suggestion
    branches: [main]
    paths: 'docs/**'
```

---

💪